apps/roam/src/utils/getSuggestiveModeConfigSettings.ts

```diff
@@ -0,0 +1,44 @@
+import { getSubTree } from "roamjs-components/util";
+import { BooleanSetting, getUidAndBooleanSetting } from "./getExportSettings";
+import { RoamBasicNode } from "roamjs-components/types";
+
+export type SuggestiveModeConfigWithUids = {
+  parentUid: string;
+  grabFromReferencedPages: BooleanSetting;
+  grabParentAndChildren: BooleanSetting;
+  pageGroups: {
+    uid: string;
+    name: string;
+    pages: { uid: string; name: string }[];
+  };
+};
+
+export const getSuggestiveModeConfigAndUids = (
+  tree: RoamBasicNode[],
+): SuggestiveModeConfigWithUids => {
+  const suggestiveModeNode = getSubTree({
+    tree,
+    key: "Suggestive Mode",
+  });
+  const pageGroupsNode = getSubTree({
+    parentUid: suggestiveModeNode.uid,
+    key: "Page Groups",
+  });
+
+  return {
+    parentUid: suggestiveModeNode.uid,
+    grabFromReferencedPages: getUidAndBooleanSetting({
+      tree: suggestiveModeNode.children,
+      text: "Include Current Page Relations",
+    }),
+    grabParentAndChildren: getUidAndBooleanSetting({
+      tree: suggestiveModeNode.children,
+      text: "Include Parent And Child Blocks",
+    }),
+    pageGroups: {
+      uid: pageGroupsNode?.uid || "",
+      name: "Page Groups",
```

Comment:
Where is this used?  rm if not necessary.

Reply:
it will be used in hyde